src/warnet/deploy.py

```diff
@@ -53,13 +54,60 @@ def deploy(directory):
 
     if (directory / NETWORK_FILE).exists():
         deploy_network(directory)
+        # deploy_fork_observer(directory)
     elif (directory / NAMESPACES_FILE).exists():
         deploy_namespaces(directory)
     else:
         click.echo(
             "Error: Neither network.yaml nor namespaces.yaml found in the specified directory."
         )
 
+def deploy_fork_observer(directory: Path):
+    network_file_path = directory / NETWORK_FILE
+    with network_file_path.open() as f:
+        network_file = yaml.safe_load(f)
+
+    namespace = get_default_namespace()
+    cmd = f"{HELM_COMMAND} 'fork-observer' {FORK_OBSERVER_CHART} --namespace {namespace}"
+
+    temp_override_file_path = ""
+    override_string = ""
+
+    # Add en entry for each node in the graph
+    # TODO: should this be moved into a chart, and only have substituted name and rpc_host values
+    for i, node in enumerate(network_file["nodes"]):
+        node_name = node.get("name")
+        node_config = f"""
+[[networks.nodes]]
+id = {i}
+name = "{node_name}"
+description = "A node. Just A node."
+rpc_host = "{node_name}"
+rpc_port = 18443
+rpc_user = "forkobserver"
+rpc_password = "tabconf2024"
```

Comment:
note to future self: we will want to move these hardcoded values out of the python code. today, if you update the tank configs to use a different port and forget to update the port value here, fork observer is borked.

even worse, the user can change the port via the yaml config (no code change required), but then to update fork observer would require a PR!

Reply:
created https://github.com/bitcoin-dev-project/warnet/issues/511 to track discussion on this

Reply:
better yet, since we parse the `network.yaml` first in `deploy_network()` we can:

1. dynamically add `rpcauth=forkobserver` to each tank only if needed (and with unique passwords too)
2. get the actual port from each node configuration to match in the FO config.
3. `description` could be more dynamic as well (tank name? version? uacomment?)